### PR TITLE
Get volunteer matches

### PR DESCRIPTION
**Pull request by apackin**
#### What's this PR do?
Retrieves information for all volunteers that are interested in VM and adds them to a MailChimp List
#### How should this be manually tested?
1) Add global.Keys and .config
2) run the file
3) check that the .csv was created correctly
4) users listed in the csv should all be present in MailChimp List: https://us5.admin.mailchimp.com/lists/members/?id=189493#p:1-s:100-so:false

#### What are the relevant tickets?
https://admin.crisistextline.org/jira/secure/RapidBoard.jspa?rapidView=5&view=detail&selectedIssue=INT-323
https://admin.crisistextline.org/jira/secure/RapidBoard.jspa?rapidView=5&view=detail&selectedIssue=INT-409

### CODE REVIEW

File: config.js [hunk 1]
```diff
@@ -110,6 +113,7 @@ CONFIG.time_interval = {
   recur_and_publish_shifts_cron_job_string: '0 */1 * * * *',
   // runs every day
   pending_users: '* * * */1 * *',
+  request_VolunteerMatches: '* * * */1 * *',
```
Comment (mbogor) [inline config.js]: '0 0 0 */1 * *' is the correct syntax

File: jobs/recruitment/volunteerMatches.js [hunk 2]
```diff
@@ -0,0 +1,157 @@
+'use strict';
+const CronJob = require('cron').CronJob;
+const request = require('request-promise');
+const fs = require('fs');
+
+new CronJob(CONFIG.time_interval.request_VolunteerMatches, function () {
```
Comment (mbogor) [inline jobs/recruitment/volunteerMatches.js]: handleVMCSVRetrieval instead, don't fall victim to peer pressure

File: jobs/recruitment/volunteerMatches.js [hunk 3]
```diff
@@ -0,0 +1,157 @@
+'use strict';
```
Comment (mbogor) [inline jobs/recruitment/volunteerMatches.js]: add comments as to what this job does, for easier ~ understanding

File: jobs/recruitment/volunteerMatches.js [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+  .then(getVolunteerMatchTSV)
+  .then(TSVtoCSV)
+  .then(CSV => {
+    return Promise.all([
+                       ...postToMailchimp(CSV), 
+                       exportCSV(CSV)
+                       ]);
+  })
+  .then(returnArr => {
+    CONSOLE_WITH_TIME('returnArr', returnArr[returnArr.length-1]);
```
Comment (mbogor) [inline jobs/recruitment/volunteerMatches.js]: ummm we can use `${}` ecma6 ftw

**Comment by mbogor** [comment]
👍

File: jobs/recruitment/volunteerMatches.js [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+      'merge_fields': removeDups[email]
+    });
+  }
+
+  return results;
+}
+
+function exportCSV (CSVString) {
+  return new Promise(function(resolve, reject) {
+    fs.writeFile('./VMVolunteers.csv', CSVString, 'utf8', (err) => {
```
Comment (tongxiang) [inline jobs/recruitment/volunteerMatches.js]: comment this out for now

**Comment by tongxiang** [comment]
👍

**Pull request merged**